From fec87866678f4cb8b5d555f6abff70dc6853709e Mon Sep 17 00:00:00 2001
From: Esun Kim <veblush@google.com>
Date: Thu, 18 Jan 2024 09:05:59 -0800
Subject: [PATCH 10/10] [Clean-up] Fix Abseil-usage (#35590)

Some minor fixes for https://github.com/grpc/grpc/pull/35589

Closes #35590

COPYBARA_INTEGRATE_REVIEW=https://github.com/grpc/grpc/pull/35590 from veblush:abseil-fix a89b800739938e482661357bb6e5c9df54d84e47
PiperOrigin-RevId: 599524969
---
 src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc      | 4 ++++
 .../lib/event_engine/windows/native_windows_dns_resolver.cc   | 1 +
 src/core/lib/gprpp/windows/directory_reader.cc                | 1 +
 test/cpp/util/channelz_sampler.cc                             | 1 +
 4 files changed, 7 insertions(+)

diff --git a/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc b/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc
index 74f7392af9..0b14723e21 100644
--- a/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc
+++ b/src/core/lib/event_engine/cf_engine/cfstream_endpoint.cc
@@ -18,6 +18,10 @@
 #include <AvailabilityMacros.h>
 #ifdef AVAILABLE_MAC_OS_X_VERSION_10_12_AND_LATER
 
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_format.h"
+
 #include "src/core/lib/event_engine/cf_engine/cfstream_endpoint.h"
 #include "src/core/lib/event_engine/trace.h"
 #include "src/core/lib/gprpp/strerror.h"
diff --git a/src/core/lib/event_engine/windows/native_windows_dns_resolver.cc b/src/core/lib/event_engine/windows/native_windows_dns_resolver.cc
index a00ed40225..ab318238c0 100644
--- a/src/core/lib/event_engine/windows/native_windows_dns_resolver.cc
+++ b/src/core/lib/event_engine/windows/native_windows_dns_resolver.cc
@@ -20,6 +20,7 @@
 
 #include <string>
 
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 
 #include <grpc/event_engine/event_engine.h>
diff --git a/src/core/lib/gprpp/windows/directory_reader.cc b/src/core/lib/gprpp/windows/directory_reader.cc
index 790e213e78..3d45ee6532 100644
--- a/src/core/lib/gprpp/windows/directory_reader.cc
+++ b/src/core/lib/gprpp/windows/directory_reader.cc
@@ -27,6 +27,7 @@
 #include <vector>
 
 #include "absl/status/statusor.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 
 #include <grpc/support/log.h>
diff --git a/test/cpp/util/channelz_sampler.cc b/test/cpp/util/channelz_sampler.cc
index 60c0e126de..db9c600c51 100644
--- a/test/cpp/util/channelz_sampler.cc
+++ b/test/cpp/util/channelz_sampler.cc
@@ -21,6 +21,7 @@
 
 #include <cstdlib>
 #include <fstream>
+#include <iomanip>
 #include <iostream>
 #include <memory>
 #include <ostream>
